cpp/src/fil/common.cuh

```diff
@@ -46,8 +46,9 @@ struct storage_base {
 };
 
 /** represents a dense tree */
-template <typename real_t>
-struct tree<dense_node<real_t>> : tree_base {
+template <typename real_t_>
+struct tree<dense_node<real_t_>> : tree_base {
+  using real_t = real_t_;
```

Comment:
I'd slightly prefer we spell out member types as e.g. `real_type` for better consistency with STL conventions, but let's not hold up the PR if that's going to be a difficult refactor.

Reply:
Done (in type-scope using declarations).